UX Improvement during Syncing and Showing File Paths

VarunT11 opened the pull request:
**Issue Reference** : https://github.com/odk-x/tool-suite-X/issues/131

**Small Description of Changes Performed** : 
- Added the Android Material Library in the _**build.gradle**_ File of the App Module of the Project.
- Replaced the Normal Alert Dialog with the Material Alert Dialog in _**AlertDialogFragment**_.
- Added a new Layout Resource File describing the Custom View of the Progress Indicator.
- Removed the _ProgressDialog_ from _**ProgressDialogFragment**_ and instead implemented Material Progress Indicator.

**Issues Resolved/Feature Update**:
- Improving User-Interface and User Experience by using Android Material Components.
- Users would now be able to see the progress of a Process using Progress Indicator.

**Screenshots**:
Earlier:
![Screenshot_20210315-003048](https://user-images.githubusercontent.com/58032608/111529775-8ca72b80-8788-11eb-9de6-840f35ed64be.jpg)
Now:
![Screenshot_20210318-001541](https://user-images.githubusercontent.com/58032608/111529727-7bf6b580-8788-11eb-967f-eddf2e52143a.jpg)
![Screenshot_20210318-001739](https://user-images.githubusercontent.com/58032608/111529770-8add6800-8788-11eb-8536-1e7dbd652aba.jpg)

**PS**: @wbrunette @linl33 Please review this PR and suggest any required changes!

linl33 commented:
runtests

VarunT11 commented:
**Small Description of Changes Performed**:
- Added the **MultiDex** Support in the _**build.gradle**_ file of the Library Module of the Application
- The previous commit was ignoring the implementation of Positive, Negative and Neutral Buttons of the **_ProgressDialogFragment._** This commit fixed that problem by restoring the **_onCreateDialog()_** method and replacing the **_ProgressDialog_** with **_MaterialAlertDialog_** with a Custom View to show the _**LinearProgressIndicator**_.

wbrunette commented:
@VarunT11 I am curious why you added the MultiDex support to the library as normally it's in the application. Did you find some documentation that you can share?

VarunT11 commented:
@wbrunette When I was going through the output logs of the Build Test that failed yesterday, I saw that the Build encountered some problems with the error - Exceeding 64k methods.
I researched about it and found this https://developer.android.com/studio/build/multidex on the Official Website of Google. Accordingly I added the MultiDex support to the Library to avoid Build from failing.

wbrunette commented:
@VarunT11 thanks for the information. It's slightly strange that we are that high on base library. 

I will look into the multi-dex more, I appreciate your diligence. You are doing everything correctly, I will investigate ways to drop the total number of methods or discuss other changes with the PMC/TSC. 

I will finish my review of your PR soon.

VarunT11 commented:
@wbrunette I just cross-checked the _**build.gradle**_ Files of the App Module of the 3 Mobile Applications - _**Services, Survey and Tables**_ and found that they already have **MultiDex** enabled in them. Thus using this library will not lead to any Build related problems in any of these applications.
However, the Build may fail if we use the usual _**runtests**_ on this PR.

wbrunette commented:
@VarunT11 you are correct that Services, Survey, and Tables are multi-dex already. However, there is also Sensors, Scan, Submit, and likely Notify (Skunkworks) that are dependent on the library. I am satisfied with your work on this. I just need more time to check the effects other downstream dependencies that are works in progress. 

You should definitely move on to other issues as I consider this complete on your end. I like your solution, it is just being careful with the down-streams.

linl33 commented on androidlibrary_lib/src/main/java/org/opendatakit/fragment/AlertDialogFragment.java:
```diff
@@ -165,14 +168,17 @@ private void setTitle(String newTitle) {
-          .setIcon(android.R.drawable.ic_dialog_info).setTitle(title).setMessage(message)
-          .setCancelable(false).setPositiveButton(getString(R.string.ok), this).create();
-      dlg.setCanceledOnTouchOutside(false);
+      androidx.appcompat.app.AlertDialog alertDialog=new MaterialAlertDialogBuilder(getActivity(), R.style.Theme_MaterialComponents_Light_Dialog_Alert)
```
Since you're using the AndroidX dialog here, the one in `getAlertDialog` of this class should also be changed. Otherwise there would be a crash on screen rotation.

linl33 commented on androidlibrary_lib/src/main/res/layout/progress_indicator_layout.xml:
```diff
@@ -0,0 +1,37 @@
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
```
The height is too short when this is used for the initialization fragment, some text is cropped. Please see attached screenshot
![Screenshot_1616481404](https://user-images.githubusercontent.com/8204095/112103948-c650ab80-8b67-11eb-8575-ff0fd1803ccd.png)

linl33 reviewed (changes requested):
@VarunT11 Thanks for the pull request!

VarunT11 commented:
**Small Description of Changes Performed**:
- Replaced `setProgress` with `setProgressCompat` in _**ProgressDialogFragment**_.
- Removed the `android:text="Message"` attribute from `TextView` in _**progress_indicator_layout**_.
- Replaced the `AlertDialog` with `Android X AlertDialog` in _**AlertDialogFragment**_.
- Changed the `android:layout_height` attribute of `TextView` from `44dp` to `wrap_content` in _**progress_indicator_layout**_.

**PS**: @linl33 The Requested Changes have been performed. Please review them.

linl33 commented on androidlibrary_lib/src/main/java/org/opendatakit/fragment/AlertDialogFragment.java:
```diff
@@ -130,10 +133,10 @@ public boolean dismissWasCalled() {
    public boolean createDialogCalled() { return createDialogCalled; }
 
-   private AlertDialog getAlertDialog() {
+   private androidx.appcompat.app.AlertDialog getAlertDialog() {
```
Now that the regular `AlertDialog` is not used, you could remove its import and import the AndroidX `AlertDialog`.

VarunT11 commented:
@linl33 I have made the required changes!

linl33 commented:
@wbrunette I would add androidcommon to the list of projects to check as well.
I'm hitting the 64k limit on androidcommon (`:androidcommon_lib:mergeDexSnapshotUitestDebugAndroidTest`) when running `connectedSnapshotUitestDebugAndroidTest` from Tables.

wbrunette commented:
@VarunT11 as an FYI, I am working on removing methods from androidlibrary to try to try to come in udner 64K. Your PR will be merged after other pending merges. Thank you for your efforts

VarunT11 commented:
@wbrunette Since you are working to remove the methods, shall I remove the **MultiDex** dependency which I added in the previous commits?

wbrunette commented:
@VarunT11 please leave it in for now as it's a complete change until we find out we can adjust the change.

VarunT11 commented:
@wbrunette Okay Sure! Thank You

wbrunette commented on androidlibrary_lib/build.gradle:
```diff
@@ -29,6 +29,7 @@ android {
         versionCode(releaseVersionCode)
         versionName(versionCodeName)
         testInstrumentationRunner(instrumentationRunner)
+        multiDexEnabled true
```
You should be able to remove this after merging in the development branch.

wbrunette commented on androidlibrary_lib/build.gradle:
```diff
@@ -107,7 +108,8 @@ dependencies {
     implementation 'androidx.fragment:fragment:1.3.0-beta01'
-
+    implementation 'com.google.android.material:material:1.4.0-alpha01'
+    implementation 'androidx.multidex:multidex:2.0.1'
```
You should be able to remove this after merging in the development branch.

wbrunette commented on androidlibrary_lib/build.gradle:
```diff
@@ -107,7 +108,8 @@ dependencies {
     implementation 'androidx.preference:preference:1.1.1'
     implementation 'androidx.fragment:fragment:1.3.0-beta01'
-
+    implementation 'com.google.android.material:material:1.4.0-alpha01'
```
Please use the last stable version 1.3.0. I believe you are using things that work with the last stable release.

wbrunette commented:
@VarunT11 after merging in changes to the development branch, then merging your branch in I was able to remove the multi-dex. Therefore, I believe you can make these changes and test on your end. After that I will merge the PR. Let us know if you run into any issues.

VarunT11 commented:
@wbrunette I have made the required changes! Please review it.

wbrunette commented:
runtests
